ipapython/install/core.py

```diff
@@ -146,15 +150,19 @@ def validate(self, value):
         return cls
 
 
-def knob(type_=_missing, default=_missing, bases=_missing, sensitive=_missing,
-         deprecated=_missing, description=_missing, cli_names=_missing,
-         cli_deprecated_names=_missing, cli_metavar=_missing):
-    if type_ is None:
-        type_ = NoneType
+_type = type
+
+
+def _knob(type=_missing, default=_missing, bases=_missing, _order=_missing,
+          sensitive=_missing, deprecated=_missing, description=_missing,
+          group=_missing, cli_names=_missing, cli_deprecated_names=_missing,
+          cli_metavar=_missing):
+    if type is None:
```

Comment:
Because the `Knob` attribute name is `type`. IMHO shadowing would be an issue only if `type` was called in the function, which is not the case here. BTW `type` is used as an argument name even in the Python standard library, on numerous occasions.